extensions/spark/kyuubi-extension-spark-3-5/src/main/scala/org/apache/kyuubi/sql/KyuubiSparkSQLCommonExtension.scala

```diff
@@ -42,6 +42,7 @@ object KyuubiSparkSQLCommonExtension {
     extensions.injectPostHocResolutionRule(InsertZorderBeforeWritingHive33)
     extensions.injectPostHocResolutionRule(FinalStageConfigIsolationCleanRule)
 
+    extensions.injectQueryStagePrepRule(DynamicShufflePartitions)
     extensions.injectQueryStagePrepRule(_ => InsertShuffleNodeBeforeJoin)
```

Comment:
shall we put the new rule behind the existed rule ?

Reply:
It seems necessary, `InsertShuffleNodeBeforeJoin` rule may introduce exchange.